src/coreclr/tests/src/tracing/eventpipe/reverseouter/reverseouter.cs

```diff
@@ -0,0 +1,118 @@
+                                });
+                            Logger.logger.Log("Starting EventPipeSession over standard connection");
+                            using Stream stream = EventPipeClient.CollectTracing(pid, config, out var sessionId);
+                            Logger.logger.Log($"Started EventPipeSession over standard connection with session id: 0x{sessionId:x}");
+                            using var source = new EventPipeEventSource(stream);
+                            Task readerTask = Task.Run(() => source.Process());
+                            await Task.Delay(500);
+                            Logger.logger.Log("Stopping EventPipeSession over standard connection");
+                            EventPipeClient.StopTracing(pid, sessionId);
+                            await readerTask;
+                            Logger.logger.Log("Stopped EventPipeSession over standard connection");
+                        }
+                        finally
+                        {
+                            mre.Set();
+                        }
+                    });
+
+                    Task reverseTask = Task.Run(async () => 
+                    {
+                        while (!mre.WaitOne(0))
+                        {
+                            var ad1 = await ReverseServer.CreateServerAndReceiveAdvertisement(serverName);
+                            Logger.logger.Log(ad1.ToString());
+                        }
+                    });
+
+                    await Task.WhenAll(reverseTask, regularTask);
+                }
+            );
+
+            fSuccess &= await subprocessTask;
+
+            return fSuccess;
+        }
+
+        public static async Task<int> Main(string[] args)
+        {
+            if (args.Length >= 1)
+            {
+                Console.Out.WriteLine("Subprocess started!  Waiting for input...");
+                var input = Console.In.ReadLine(); // will block until data is sent across stdin
+                Console.Out.WriteLine($"Received '{input}'.  Exiting...");
+                return 0;
+            }
+
+            bool fSuccess = true;
+            if (!IpcTraceTest.EnsureCleanEnvironment())
+                return -1;
+            IEnumerable<MethodInfo> tests = typeof(ReverseValidation).GetMethods().Where(mi => mi.Name.StartsWith("TEST_"));
+            foreach (var test in tests)
+            {
+                Logger.logger.Log($"::== Running test: {test.Name}");
+                bool result = true;
+                try
+                {
+                    result = await (Task<bool>)test.Invoke(null, new object[] {});
+                }
+                catch (Exception e)
+                {
+                    result = false;
+                    Logger.logger.Log(e.ToString());
+                }
+                fSuccess &= result;
+                Logger.logger.Log($"Test passed: {result}");
```

Comment:
Won't this always print "test passed" even when it didn't?

Reply:
It should print `Test passed: false` on test failure.